### ISSUE TITLE
Rendering corrupt latex expression can cause browser to hang!

### ISSUE THREAD

**Issue by ajhyndman** [mathquill/mathquill#775]
I haven't yet investigated the direct cause, but if the following expression is pasted into a MathQuill input, rendering code appears to get caught in unresolved recursion.  The result is that the browser will become unrecoverably unresponsive!

```\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\su```

**Comment by laughinghan** [comment]
Interesting. Bisecting to about this length:

```
\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\sum_{\su
```

It hangs for a few seconds but then comes back. Presumably there's some kind of exponential growth of something, so at the length that you're pasting in it's effectively an infinite loop.

**Comment by ajhyndman** [comment]
Yeah, it does seem like MathQuill is performing terminating computation, it just becomes prohibitively expensive at some point.
